### PR TITLE
Week overview edit task

### PR DESCRIPTION
**Pull request by DanielRBech**
## Description

Creating of edit task in the week overview. Creation of a TimeCalender component for selection of hours and minutes, and a NormalCalender for date selection. Also createdthe imagePickerSelector, which takes a picture from the users photo libary, in addition with a function to limit what the image can be (Ratio, quality, mediatype).

## Type of PR

- Feature

https://github.com/aau-giraf/weekplan/issues/25

Please link the related issues/tickets here.

## Checklist

- [ X] Code follows the code guidelines
- [X ] Tests have been added/updated (if applicable)
- [x] Documentation has been updated (if applicable)

### CODE REVIEW

File: components/EditTaskButton.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+type EditTaskButtonProps = {
+    label: string,
+};
+
+const EditTaskButton: React.FC<EditTaskButtonProps> = (props) => {
+    const [showModal, setShowModal] = React.useState(false);
+    const [newTitle, setNewTitle] = React.useState(props.label);
+    const [startTime, setStartTime] = React.useState(new Date());
+    const [endTime, setEndTime] = React.useState(new Date());
```
Comment (MertzAndreas) [inline components/EditTaskButton.tsx:14]: Import useState direkte. Lav de te useState funktioner til et enkelt som bruger et objekt.

File: components/NormalCalender.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { View, Text } from "react-native";
+import RNDateTimePicker from "@react-native-community/datetimepicker";
+
+type CalenderProps = {
+    minDate: Date,
+}
+
+const NormalCalender: React.FC<CalenderProps> = ({ minDate }) => {
+    const [date, setDate] = React.useState(new Date());
```
Comment (MertzAndreas) [inline components/NormalCalender.tsx:10]: Også import useState her.

File: components/TimeCalender.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        if (selectedDate) {
+            setEndDate(selectedDate);
+            onTimeChange(startDate, selectedDate);
+        }
+    };
+
+    return (
+        <View>
+            <View style={styles.pickerContainer}>
+                {/* Time pickers in hour/minute form*/}
```
Comment (MertzAndreas) [inline components/TimeCalender.tsx:30]: Skriv en kommentar længere oppe som forklare hvad komponenten skal

File: components/WeekdayHeader.tsx
Comment (MertzAndreas) [inline components/WeekdayHeader.tsx:1]: I fremtiden så lad vær med at tilføje disse i commits

File: utils/pickImage.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import * as ImagePicker from 'expo-image-picker';
+
+const pickImage = async () => {
+    //The result object contains information about the selected image
```
Comment (MertzAndreas) [inline utils/pickImage.ts:4]: Flyt comment til toppen

File: __tests__/pickImige.test.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import * as ImagePicker from 'expo-image-picker';
+
+jest.mock('expo-image-picker', () => ({
+    launchImageLibraryAsync: jest.fn(),
+    MediaTypeOptions: {
+        Images: 'Images',
+    },
+}));
+
+describe('pickImage', () => {
```
Comment (MertzAndreas) [inline __tests__/pickImige.test.ts:11]: Not very descriptive.

**Review by MertzAndreas — CHANGES_REQUESTED** [review]

**Review by MertzAndreas — APPROVED** [review]

File: components/EditTaskButton.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import React, {useState} from "react";
+import { View, Button, Modal, StyleSheet, ScrollView, TextInput, Text, Pressable } from "react-native";
+import TimeCalender from "./TimeCalender";
+import ImagePickerSelector from "./ImagePickerSelector";
+
+type EditTaskButtonProps = {
+    label: string,
+};
+
+type SumbitProps = {
```
[the remark this answers is not in the document]
Comment (DanielRBech) [reply components/EditTaskButton.tsx:10]: Yeps thanks i missed

**Review by LokeSGJ — APPROVED** [review]

File: components/EditTaskButton.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    });
+    const handleTimeChange = (start: Date, end: Date) => {
+        setSubmitProps({
+            title: submitProps.title,
+            startTime: start,
+            endTime: end,
+        });
+    };
+
+    const handleSubmit = () => {
```
Comment (PixieStick314) [inline components/EditTaskButton.tsx:32]: I'm assuming that this is temporary

Comment (DanielRBech) [reply components/EditTaskButton.tsx:32]: Yes, when the endpoint is established, it will be used as an update activity

File: utils/pickImage.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+//The result object contains information about the selected image, such as the URI of the image file.
+const pickImage = async () => { 
+    let result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.Images,
+        allowsEditing: true,
+        aspect: [4, 3],
+        quality: 1,
+    });
+
+    console.log(result);
```
Comment (PixieStick314) [inline utils/pickImage.ts:12]: Is this for testing purposes, or is it because the backend isn't set up?

**Review by PixieStick314 — APPROVED** [review]
It's only the "console.log" lines that are out of place, but I'm choosing to assume that they are temporary implementations due to the lacking backend

File: utils/pickImage.ts (hunk 8 above)
Comment (DanielRBech) [reply utils/pickImage.ts:12]: To see what a image is composed of for future delvelopment (type like png/jpeg, size, location ect.)

**Pull request merged**